generate/generate.js

```diff
@@ -0,0 +1,253 @@
+        exercise,
+        functions,
+        testCase
+      );
+      generatedChunks.push(
+        `
+  ${toElmComments(testCase.comments)}
+
+  describe "${testCase.description}" [ ${subtestsOutput} ]
+  `
+      );
+    } else {
+      generatedChunks.push(generateTestCode(exercise, functions, testCase));
+    }
+  }
+  return generatedChunks.join(", ");
+}
+
+// Generate the test code for one test case
+function generateTestCode(exercise, functions, testCase) {
+  // Extract the expected value
+  let expectedValue = testCase.expected;
+
+  // If the function can error, adjust the expected value
+  if (functions[testCase.property].canError) {
+    if (expectedValue.hasOwnProperty("error")) {
+      expectedValue = "Err " + jsonValueToElm(expectedValue.error);
+    } else {
+      expectedValue = "Ok " + jsonValueToElm(expectedValue);
+    }
+    // Otherwise just convert it into a string
+  } else {
+    expectedValue = jsonValueToElm(expectedValue);
+  }
+
+  // Stringify the test case inputs
+  let inputs = "";
+  for (const [key, value] of Object.entries(testCase.input)) {
+    inputs += " " + jsonValueToElm(value);
+  }
+
+  // Generate a re-implement message if this tests re-implements another one
+  let reimplements = [];
+  if (testCase.reimplements) {
+    reimplements = [
+      "This test reimplements the test with uuid " + testCase.reimplements,
+      "Please identify that test and remove it. Link:",
+      "https://github.com/exercism/problem-specifications/blob/main/exercises/" +
+        testCase.reimplements +
+        "/canonical-data.json",
+    ];
+  }
+
+  return `
+  ${toElmComments(reimplements)}
+  ${toElmComments(testCase.comments)}
+
+  -- skip <|
+  test "${testCase.description}" <|
+    \\() ->
+      ${exercise}.${testCase.property} ${inputs}
+        |> Expect.equal (${expectedValue})`;
+}
+
+// Convert the list of strings composing the comment in the canonical data
+// into Elm comments by prepending -- to all lines.
+function toElmComments(dataComments) {
+  if (dataComments) {
+    return dataComments.map((line) => "  -- " + line).join("\n");
+  } else {
+    return "";
+  }
+}
+
+// Convert a JSON value into as close as possible Elm code string
+function jsonValueToElm(json) {
+  switch (typeof json) {
+    case "boolean":
+      return json ? "True" : "False";
+    case "number":
+      return JSON.stringify(json);
+    case "string":
+      return JSON.stringify(json);
+    case "object":
+      if (json == null) {
+        return "Nothing";
+      } else if (json.constructor === Array) {
+        // Apply a recursive call to all elements of the array
+        const arrayContent = json.map(jsonValueToElm).join(", ");
+        return `[ ${arrayContent} ]`;
+      } else {
+        // Apply a recursive call to all attribute values,
+        // and lowercase the object keys.
+        const toElmKey = (key) =>
+          replaceReservedWord(key[0].toLowerCase() + key.slice(1));
```

Comment:
Setting the first character to lower case is a good call! I haven't seen variables with uppercase, but it doesn't mean there aren't any.

Reply:
Yes I saw that when trying the script on the nucleotide count canonical data where the object keys are uppercase "A", "G", "T", "C".